liquibase-core/src/main/java/liquibase/resource/PathResource.java

```diff
@@ -35,13 +35,36 @@ public InputStream openInputStream() throws IOException {
         }
     }
 
+    @Override
+    public boolean exists() {
+        return Files.exists(path);
+    }
+
+    @Override
+    public Resource resolve(String other) {
+        return new PathResource(resolvePath(other), path.resolve(other));
+    }
+
+    @Override
+    public Resource resolveSibling(String other) {
+        return new PathResource(resolveSiblingPath(other), path.resolveSibling(other));
+    }
+
     @Override
     public boolean isWritable() {
         return !Files.isDirectory(this.path) && Files.isWritable(path);
     }
 
     @Override
-    public OutputStream openOutputStream() throws IOException {
+    public OutputStream openOutputStream(boolean createIfNeeded) throws IOException {
+        if (!exists()) {
+            if (createIfNeeded) {
+                path.getParent().toFile().mkdirs();
+            } else {
+                throw new IOException("File " + this.getUri() + " does not exist");
+            }
+        }
+
```

Comment:
These changes are fine. I just want to point out that `Files.newOutputStream(this.path)` will truncate the existing file before writing to it. I found this behavior a bit surprising originally but figured it was intentional. Now, with the new `createIfNeeded` parameter, this behavior is much more surprising. If it's intentional and desired, let's call it out in the documentation.

Reply:
It seems like the truncating is the behavior we want? I was a bit worried that we'd have to add logic to do that, but sounds like we're not?

I think the expectation is "openOutputStream lets you write what you want for content to the file" and not have to worry about "if you write less bytes than the current file has, the final result is a combination of what you wrote and what was there before". 

But why are you thinking that especially with the createIfNeeded it's extra surprising? I'm maybe not following?

Reply:
Truncating probably is the behavior we want. I just think it needs to be called out expressly in the documentation, because it might not be obvious to a user. It would be confusing to a user who thinks to themselves "I told you not to create the file, why are you overwriting the entire thing".

> I think the expectation is "openOutputStream lets you write what you want for content to the file" and not have to worry about "if you write less bytes than the current file has, the final result is a combination of what you wrote and what was there before".

I can see where you're going here. I can also see a not insignificant number of users who are surprised by the fact that "opening" an output stream to a file also erases the entire file contents.